catalyst/utils/typing.py

```diff
@@ -1,12 +1,23 @@
-from typing import Union  # isort:skip
+from typing import Union, Dict  # isort:skip
 
 import torch
 from torch import nn, optim
+from torch.utils import data
 
-Model = nn.Module
+Model = Union[nn.Module, Dict[str, nn.Module]]
 Criterion = nn.Module
-Optimizer = optim.Optimizer
+Optimizer = Union[optim.Adam, optim.SGD]
 Scheduler = optim.lr_scheduler._LRScheduler  # noinspection PyProtectedMember
+Dataset = data.Dataset
```

Comment:
this is very good :)

Reply:
Syntax sugar to make beautiful catalyst typing :)
Ill change it just give me a word. :D

Reply:
i truly think, adding `Dataset` and `DataLoader` to our typing was a pretty good idea :)

Reply:
I also like this, but are you sure that it will be clear for everyone? I mean, `torch.utils.data.Dataset` is pretty much obvious, but `catalyst.utils.typing.Dataset` from the first look (without looking how it was defined) seems like some catalyst-internal special dataset subclass. Will it not look confusing?

So, I like this, but I am afraid it may look confusing. But if you disagree with me it's ok, let it be

Reply:
I'll leave it and we'll see.
For me, it's not more confusing than Model that is nn.Module :)
I think we should do it one way or another for majority of typing entities.